File: backend/internal/database/models/job.model.go
package models

import "github.com/gianghp123/Vidmerce/backend/internal/core/enums"

type JobEntity struct {
	BaseItem
	// Creation timestamp
	CreatedAt string `dynamodbav:"createdAt"`
	// Error log if failed
	ErrorLog *string `dynamodbav:"errorLog,omitempty"`
	// JobEntity payload data
	Payload map[string]interface{} `dynamodbav:"payload,omitempty"`
	Status  enums.JobStatus        `dynamodbav:"status"`
	// Target entity ID
	TargetID *string       `dynamodbav:"targetId,omitempty"`
	Type     enums.JobType `dynamodbav:"type"`
}

File: web/src/types/models/job.model.ts
import { JobStatus } from "../enums/job-status.enum";
import { JobType } from "../enums/job-type.enum";

export interface IJob {
  id: string;
  /**
   * Creation timestamp
   */
  createdAt: string;
  /**
   * Error log if failed
   */
  errorLog?: string;
  /**
   * IJob payload data
   */
  payload?: { [key: string]: any };
  status: JobStatus;
  /**
   * Target entity ID
   */
  targetId?: string;
  type: JobType;
}
